# Week 6. Eigenvalues and eigenvectors, Leslie matrix

#### A. Finding eigenvalues and eigenvectors using Julia

In [3]:
A=[1 0 3; 1 0 -1;0 0 2]

3×3 Array{Int64,2}:
 1  0   3
 1  0  -1
 0  0   2

In [10]:
# eigenvalues of A, 
eig([1 0 3;1 0 -1;0 0 2])

([0.0,1.0,2.0],
[0.0 0.707107 0.904534; 1.0 0.707107 0.301511; 0.0 0.0 0.301511])

In the above, the 1st output line $[0.0,\ 1.0,\ 2.0]$ contains the eigenvalues of A: $\lambda_1=0$, $\lambda_2=1$ and $\lambda_3=2$. 
The the 2nd output line $[0.0,\ 0.707107,\ 0.904534;\ 1.0,\ 0.707107,\ 0.301511;\ 0.0,\ 0.0,\ 0.301511]$ contains 3 column vectors $v_1$, $v_2$ and $v_3$: $[v_1,\; v_2,\; v_3]=$

In [11]:
[0.0 0.707107 0.904534; 1.0 0.707107 0.301511; 0.0 0.0 0.301511]

3×3 Array{Float64,2}:
 0.0  0.707107  0.904534
 1.0  0.707107  0.301511
 0.0  0.0       0.301511

So, $v_1$, $v_2$ and $v_3$ are the 1st, 2nd and 3rd column of the above matrix. 

#### B. One can find eigenvalues and eigenvectors separately

In [8]:
eigvals(A)

3-element Array{Float64,1}:
 0.0
 1.0
 2.0

In [14]:
eigvecs(A)

3×3 Array{Float64,2}:
 0.0  0.707107  0.904534
 1.0  0.707107  0.301511
 0.0  0.0       0.301511

Though the eigenvalues and eigenvectors are obtained separately (and needs more input), the outputs are clearer than that from the "eig()" function.

#### C. Leslie matrix 

Determine whether the population will survive if the leslie matrix for a structured population model is the following matrix $B$.

In [15]:
B=[0 0.3 0.7 1.2 3 2;0.6 0 0 0 0 0;0 0.7 0 0 0 0;0 0 0.8 0 0 0;0 0 0 0.85 0 0;0 0 0 0 0.55 0]

6×6 Array{Float64,2}:
 0.0  0.3  0.7  1.2   3.0   2.0
 0.6  0.0  0.0  0.0   0.0   0.0
 0.0  0.7  0.0  0.0   0.0   0.0
 0.0  0.0  0.8  0.0   0.0   0.0
 0.0  0.0  0.0  0.85  0.0   0.0
 0.0  0.0  0.0  0.0   0.55  0.0

In [17]:
eigvals(B)

6-element Array{Complex{Float64},1}:
     1.182+0.0im     
  0.268734+0.900901im
  0.268734-0.900901im
 -0.647047+0.536928im
 -0.647047-0.536928im
 -0.425369+0.0im     

There are two real eigenvalues (the 1st and last elenents in the above array). The largest of them is $\lambda=1.182>1$. Thus, the population increases over time and hence will survive.